clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -86,6 +85,25 @@ public AbortedTransaction(long pid, long firstOffset) {
         }
 
         @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            AbortedTransaction that = (AbortedTransaction) o;
+
+            return pid == that.pid && firstOffset == that.firstOffset;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = (int) (pid ^ (pid >>> 32));
```

Comment:
We use the pattern that I am using here all over the code. You can see for yourself. If you'd like to start a conversation on what's the best way to implement equals and hashCode, that is fine, but let's not block this bug fix on that.

These are necessary so that we can do `assertEquals` in the test. If we don't do it here, we'll have to do the same in more lines in the test.